### DISCUSSION TITLE
Types appear broken and not unwrapping refs in getters

### DISCUSSION THREAD

**Discussion by silverbackdan** [vuejs/pinia#2485] [Help and Questions]
### Reproduction

https://github.com/components-web-app/cwa-nuxt-module/tree/nuxt3/src/runtime/storage/stores/resources

### Steps to reproduce the bug

I have defined a store like this:

https://github.com/components-web-app/cwa-nuxt-module/blob/nuxt3/src/runtime/storage/stores/resources/resources-store.ts

You can see the types I'm declaring here which was originally created as shorthands for types which were only internal:
https://github.com/components-web-app/cwa-nuxt-module/blob/nuxt3/src/runtime/storage/stores/cwa-store-types.d.ts

When creating a store like this e.g.:
```
const storeCls = new ResourcesStore()
const store = storeCls.useStore()
```

Typescript is now expecting me to access getters like:
```
store.findDraftComponentIri.value(args)
```

Whereas it typed it properly before and I do not need the `.value` as computed vars are unwrapped. E.g.

```
store.findDraftComponentIri(args)
```

### Expected behavior

I expected the types to be the same as before and to unwrap the getters. Am I missing a new type/interface?

### Actual behavior

Although I access the computed getter references without `.value` - the type hinting expects me to use `.value` which would actually result in a bug.

### Additional information

Hopefully it's a simple change which I've missed but I was looking a while and can't seem to see the solution yet.

Also, the type behaviour is the same if I remove the return type from the `useStore` method.

**Comment by silverbackdan** [comment]
@posva - can I ask why a discussion when it's a regression on typing?

**Reply by web-deved** [reply]
Have you managed to find a solution for this? The issue still persists for us in all stores and makes working with the IDE challenging.

**Reply by silverbackdan** [reply]
I'm afraid not - it was VERY disappointing to see a bug report on regression get converted into a discussion. It may have something to do with VueJS typing changes for unwrapping refs but really haven't been able to work it out. Typescript can get a bit above my head sometimes, not my fave for typing at all, but does prove useful... until things like this happen. Thanks for the upvote

**Reply by silverbackdan** [reply]
I believe it may be resolve with an upgrade of packages now. I'm not sure where the issue was, but after a blanket upgrade we are working again - not sure if it was Typescript, Pinia, Vue etc. etc. But it appears to be working... LMK?

**Reply by web-deved** [reply]
Okay thank you, I'll upgrade relevant packages and see if that fixes it for us as well.

**Comment by Vap0r1ze** [comment]
If anyone comes across this issue, the fix is making sure your versions of `pinia` and `vue` are compatible. For me I just updated `vue` to latest, but you may want to check the version release dates if you must use an old version.

**Comment by liana-p** [comment]
I'm getting the same issue, trying to update vue and pinia to latest doesn't work.
